luigi/static/visualiser/js/graph.js

```diff
@@ -64,18 +64,27 @@ Graph = (function() {
     /* Compute the maximum depth of each node for layout purposes, returns the number
        of nodes at each depth level (for layout purposes) */
     function computeDepth(nodes, nodeIndex) {
-        function descend(n, depth) {
-            if (n.depth === undefined || depth > n.depth) {
-                n.depth = depth;
-                $.each(n.deps, function(i, dep) {
-                    if (nodeIndex[dep]) {
-                        descend(nodes[nodeIndex[dep]], depth + 1);
-                    }
-                });
+
+        // memoized to take linear time in number of nodes and edges
+        function height(taskId) {
+            var node = nodes[nodeIndex[taskId]];
+
+            // storing height in depth here to save space
+            if (node.depth === undefined || node.depth === -1) {
+                node.depth = Math.max(0, Math.max.apply(null, node.deps.map(height)) + 1);
             }
+            return node.depth;
         }
-        descend(nodes[0], 0);
+        var max_height = height(nodes[0].taskId);
```

Comment:
So are we assuming that `nodes[0]` is a root task here? I have never read this client code. :)